spotbugs/src/main/java/edu/umd/cs/findbugs/SourceLineAnnotation.java

```diff
@@ -944,6 +944,34 @@ public String getSourcePath() {
         return sourcePath;
     }
 
+    public String getRealSourcePath() {
+        if (isSourceFileKnown()) {
+            Project project = myProject.get();
+            if (project != null) {
+                try {
+                    SourceFinder mySourceFinder = project.getSourceFinder();
+                    return new File(mySourceFinder.findSourceFile(this).getFullFileName()).getCanonicalPath();
+
+                } catch (IOException e) {
+                    assert true;
+                }
+            }
+            else {
+              AnalysisContext currentAnalysisContext = AnalysisContext.currentAnalysisContext();
+              if (currentAnalysisContext != null) {
+                try {
+                  SourceFinder sourceFinder = currentAnalysisContext.getSourceFinder();
+                  return new File(sourceFinder.findSourceFile(this).getFullFileName()).getCanonicalPath();
+                } catch (IOException e) {
+                  e.printStackTrace(System.out);
```

Comment:
`e.printStackTrace` is generally frowned upon. Can you use a `java.util.logging.Logger` instead?

Also, wrong indentation.